### PR TITLE
feat(design-system): add curated tags

### PR DESCRIPTION
**Pull request by tiborux**
Added curated tags tokens.

EX-5089

### CODE REVIEW

File: packages/x-components/src/views/design-system/design-system-tag.vue [hunk 1]
```diff
@@ -143,15 +143,38 @@
         <span>text tag extra large name</span>
       </div>
     </article>
+    <article class="x-list x-list--wrap x-list--gap-06 x-list--align-start">
```
Comment (tiborux) [inline packages/x-components/src/views/design-system/design-system-tag.vue:146]: Im not sure about this example, I wouldn't add it or mix tags and curated with different styles

Comment (diegopf) [reply packages/x-components/src/views/design-system/design-system-tag.vue:146]: IMO having them alone is enough.

File: packages/x-components/src/design-system/components/tag/default.tokens.scss [hunk 2]
```diff
@@ -35,4 +42,14 @@
...
+  // typography curated
+  --x-font-family-curated-tag-default: var(--x-font-family-text);
+  --x-size-font-curated-tag-default: var(--x-size-font-text);
+  --x-size-line-height-curated-tag-default: var(--x-size-line-height-text);
+  --x-number-font-weight-curated-tag-default: var(--x-number-font-weight-text);
+  --x-font-family-curated-tag-default-selected: var(--x-font-family-tag-default);
+  --x-size-font-curated-tag-default-selected: var(--x-size-font-tag-default);
+  --x-size-line-height-curated-tag-default-selected: var(--x-size-line-height-tag-default);
+  --x-number-font-weight-curated-tag-default-selected: var(--x-number-font-weight-tag-default);
+
```
Comment (diegopf) [inline packages/x-components/src/design-system/components/tag/default.tokens.scss:54]: remove the empty line

**Review by diegopf — CHANGES_REQUESTED** [review]

File: packages/x-components/src/design-system/components/tag/default.scss [hunk 3]
```diff
@@ -138,6 +151,19 @@
...
+      // color
+      background-color: var(--x-color-background-curated-tag-default-selected);
+      border-color: var(--x-color-border-curated-tag-default-selected);
+      color: var(--x-color-text-curated-tag-default-selected);
+
+      // typography
+      line-height: var(--x-size-line-height-tag-default-selected);
+      font-family: var(--x-font-family-tag-default-selected);
+      font-size: var(--x-size-font-tag-default-selected);
+      font-weight: var(--x-number-font-weight-tag-default-selected);
```
Comment (javieri-empathy) [inline packages/x-components/src/design-system/components/tag/default.scss:164]: Shouldn't this use the `curated` variant tokens?

File: packages/x-components/src/views/design-system/design-system-tag.vue [hunk 4]
```diff
@@ -143,15 +143,38 @@
...
+      <h2 class="x-title2">Curated</h2>
+      <div class="x-tag x-tag--is-curated">
+        <CuratedCheck />
+        <span>text tag</span>
+      </div>
+      <div class="x-tag x-tag--is-curated">
```
Comment (javieri-empathy) [inline packages/x-components/src/views/design-system/design-system-tag.vue:152]: The curated tags work with pill and card variants, but do not work with ghost variant

File: packages/x-components/src/design-system/components/tag/default.tokens.scss [hunk 5]
```diff
@@ -6,6 +6,13 @@
   --x-color-background-tag-default-selected: var(--x-color-base-neutral-95);
   --x-color-border-tag-default-selected: var(--x-color-border-tag-default);
   --x-color-text-tag-default-selected: var(--x-color-text-tag-default);
+  // color curated
+  --x-color-background-curated-tag-default: var(--x-color-base-neutral-100);
```
Comment (javieri-empathy) [inline packages/x-components/src/design-system/components/tag/default.tokens.scss:10]: Is the naming fine? `x-color-background-curated-tag-default` means that we are treating `curated-tag` as a component (and therefore it should have their own styles in its own folder) while I think `curated` is a modifier.

However treating `curated` as a modifier and naming it `x-color-background-tag-default-curated` brings the problem of combining it with the other `selected` modifier.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/design-system/components/tag/default.tokens.scss (hunk 5 above)
Comment (tiborux) [reply packages/x-components/src/design-system/components/tag/default.tokens.scss]: the name should be:

`--x-color-background-tag-default-curated-selected`
`--x-color-background-tag-default-curated`

**Review by diegopf — DISMISSED** [review]

File: packages/x-components/src/views/design-system/design-system-tag.vue (hunk 4 above)
Comment (javieri-empathy) [reply packages/x-components/src/views/design-system/design-system-tag.vue:152]: Now the ghost variant removes the background in curated RTs, but when selected it does not apply the font weight, which causes them to look identical when selected and when un-selected. You can see so in the second RT of this picture which has the ghost variant applied

image

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by diegopf — APPROVED** [review]

**Pull request merged**